openlibrary/templates/home/custom_ia_carousel.html

```diff
@@ -12,4 +12,10 @@
     "mode": "page",
     "limit": limit
   }
+
+$ web_lang= get_lang() or 'en'
+$if user_lang_only and web_lang:
+  $ user_lang= convert_iso_to_marc(web_lang)
+   $if user_lang:
+    $ query = query + ' language:' + user_lang
```

Comment:
This needs to happen higher up; the query has already been used by this point! 😁